### ISSUE TITLE
message error when using `syslog` source in `udp` mode and appname miss

### ISSUE THREAD

**Issue by optimistic9527** [timberio/vector#4811]
### Vector Version

```
docker.io/timberio/vector:0.10.0-alpine

```

### Vector Configuration File

```toml
[sources.fort]
  type = "syslog" # required
  address = "0.0.0.0:1894" # required, required when mode = "tcp" or mode = "udp"
  mode = "udp" # required
[sinks.my_sink_id]
  # General
  type = "console" # required
  inputs = ["fort"] # required
  target = "stdout" # optional, default

  # Encoding
  encoding.codec = "json" # required
```

### Debug Output

```
Oct 30 07:25:07.565  INFO vector::topology::builder: Healthcheck: Passed.
Oct 30 07:25:07.565  INFO vector::topology::builder: Healthcheck: Passed.
Oct 30 07:25:07.569  INFO vector::topology::builder: Healthcheck: Passed.
Oct 30 07:25:07.573  INFO vector::topology::builder: Healthcheck: Passed.
Oct 30 07:25:32.074 DEBUG source{name=fort type=syslog}: tokio_reactor::registration: scheduling Read for: 4
Oct 30 07:25:32.076 DEBUG sink{name=fort-out type=kafka}: vector::sinks::kafka: sending event. count=1
{"appname":"{\"username\"","facility":"local0","host":"opsaudit","hostname":"opsaudit","message":"\"admin\", \"ip\": \"10.217.59.178\", \"type\": \"\", \"user_agent\": \"Go-http-client/1.1\", \"datetime\": \"2020-10-30 15:25:31\", \"mfa\": 0, \"status\": true, \"city\": \"局域网\", \"optype\": \"user-login\"}","severity":"info","source_ip":"10.218.65.208:46898","source_type":"syslog","timestamp":"2020-10-30T15:25:32Z"}
```

### Expected Behavior

  raw log miss appname,but output parse message into appname

### Actual Behavior

message error

### Expected Data

```
{"appname":"","facility":"local0","host":"opsaudit","hostname":"opsaudit","message":"{\"username\":\"admin\", \"ip\": \"10.217.59.178\", \"type\": \"\", \"user_agent\": \"Go-http-client/1.1\", \"datetime\": \"2020-10-30 15:25:31\", \"mfa\": 0, \"status\": true, \"city\": \"局域网\", \"optype\": \"user-login\"}","severity":"info","source_ip":"10.218.65.208:46898","source_type":"syslog","timestamp":"2020-10-30T15:25:32Z"}
```
### socket Data
```
<134>Oct 30 16:05:54 opsaudit  {\"username\": \"admin\", \"ip\": \"10.217.59.178\", \"type\": \"\", \"user_agent\": \"Go-http-client/1.1\", \"datetime\": \"2020-10-30 16:05:45\", \"mfa\": 0, \"status\": true, \"city\": \"局域网\", \"optype\": \"user-login\"}
```

**Comment by jszwedko** [comment]
Opened an upstream PR to handle parsing messages with no tag. Notably, this will only work if there is an extra space, as is shown in the example, indicating the end of the tag field; otherwise the parser cannot differentiate between what is a tag and what is the beginning of the message for RFC3164 syslog messages.

**Comment by jszwedko** [comment]
@FungusHumungus would you mind taking a look at https://github.com/FungusHumungus/syslog-loose/pull/5 ?

**Comment by FungusHumungus** [comment]
@jszwedko Sure. Sorry, I totally missed those.

**Comment by FungusHumungus** [comment]
@jszwedko I've pushed a new version of the crate with your fix in,  [v0.5.0](https://crates.io/crates/syslog_loose).

**Comment by jszwedko** [comment]
Awesome, thanks!
